jwks.go

```diff
@@ -180,22 +191,84 @@ func (j *JWKS) RawJWKS() []byte {
 func (j *JWKS) ReadOnlyKeys() map[string]interface{} {
 	keys := make(map[string]interface{})
 	j.mux.Lock()
-	for kid, cryptoKey := range j.keys {
-		keys[kid] = cryptoKey.public
+	for _, key := range j.keys {
+		keys[key.kid] = key.Public
 	}
 	j.mux.Unlock()
 	return keys
 }
+func findKeyIdx(alg string, kid string, keys []ParsedJWK) int {
+	for idx, key := range keys {
+		if key.kid == kid {
+			return idx
+		}
+	}
+	return -1
+}
+
+func (j *JWKS) canUseKey(key ParsedJWK) bool {
+	canUseKey := true
+	// jwkUseWhitelist might be empty if the jwks was from keyfunc.NewJSON() or if JWKUseNoWhitelist option was true.
+	// in this case we don't restrict "use" parameter
+	if len(j.jwkUseWhitelist) > 0 {
+		_, canUseKey = j.jwkUseWhitelist[key.use]
+	}
+	return canUseKey
+}
+
+func (j *JWKS) getMatchingKeys(alg, kid, iss string) []*ParsedJWK {
+	result := make([]*ParsedJWK, 0, 1)
```

Comment:
```suggestion
	var result []*ParsedJWK
```
What you've got is fine, but I don't think it has any advantage over the simpler declaration above